document-store/src/main/java/org/hypertrace/core/documentstore/mongo/update/parser/MongoUpdateParser.java

```diff
@@ -41,6 +42,6 @@ public BasicDBObject buildUpdateClause(final Collection<SubDocumentUpdate> updat
   }
 
   private SubDocumentUpdate getLastUpdatedTimeUpdate() {
-    return SubDocumentUpdate.of(SubDocument.implicitUpdatedTime(), clock.millis());
+    return SubDocumentUpdate.of(SubDocument.implicitUpdateTime(), new Date(clock.millis()));
```

Comment:
Replacing this could result in the other timestamp not getting updated. We want both of them to be updated.

Reply:
can we return a List < SubDocumentUpdate> here ?